migrator/migrations/m_192_to_m_193_make_api_token_names_unique/migration_impl.go

```diff
@@ -0,0 +1,116 @@
+package m192tom193
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	APITokenStore "github.com/stackrox/rox/migrator/migrations/m_192_to_m_193_make_api_token_names_unique/apitokenstore"
+	newAPITokenStore "github.com/stackrox/rox/migrator/migrations/m_192_to_m_193_make_api_token_names_unique/apitokenstore/new"
+	oldAPITokenStore "github.com/stackrox/rox/migrator/migrations/m_192_to_m_193_make_api_token_names_unique/apitokenstore/old"
+	midPkgSchema "github.com/stackrox/rox/migrator/migrations/m_192_to_m_193_make_api_token_names_unique/schema/mid"
+	newPkgSchema "github.com/stackrox/rox/migrator/migrations/m_192_to_m_193_make_api_token_names_unique/schema/new"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+)
+
+const (
+	batchSize = 500
+)
+
+var (
+	seenTokenNames = make(map[string]int)
+)
+
+func migrate(database *types.Databases) error {
+	oldAPITokenStorage := oldAPITokenStore.New(database.PostgresDB)
+	newAPITokenStorage := newAPITokenStore.New(database.PostgresDB)
+	migrator := newMigrator(oldAPITokenStorage, newAPITokenStorage)
+	return migrator.doMigrate(database)
+}
+
+type migratorImpl struct {
+	oldStore APITokenStore.Store
+	newStore APITokenStore.Store
+}
+
+func newMigrator(oldStore APITokenStore.Store, newStore APITokenStore.Store) *migratorImpl {
+	return &migratorImpl{
+		oldStore: oldStore,
+		newStore: newStore,
+	}
+}
+
+func (m *migratorImpl) doMigrate(database *types.Databases) error {
+	// Create name column.
```

Comment:
nit:  May be worth expanding on the comment as to why the table update is a 2 step process.